tracer/src/Datadog.Trace/ClrProfiler/NativeMethods.cs

```diff
@@ -249,6 +249,18 @@ public static void UpdateSettings(string[] keys, string[] values)
             }
         }
 
+        public static void GetIastMetrics(out int instrumentedSources, out int instrumentedPropagations, int[] instrumentedSinks)
```

Comment:
There are some non-iast metrics that we would really like to retrieve from the native tracer too. I wonder if it might be preferable to try to generalise this, `GetNativeMetrics()` or similar. We won't be able to change this method again, and so it might be better to be reusable.

That said, I wonder if the native side is really the right place to collect these metrics 🤔  Every instrumentation calls managed code right, so wouldn't it be preferable to inject that in the managed side (for safety and simplicity).

Reply:
We need to detect every time that we modify the code for a callsite call. I would say that this can only be detected in the native side. We can later call (or not) a managed aspect that replaces string.concat and other methods  but the instrumentation itself is detected and performed in the native side.